shell/platform/android/io/flutter/embedding/FlutterFragment.java

```diff
@@ -164,7 +179,7 @@ public void onBackPressed() {
    * @param grantResults permission grants or denials
    */
   public void onRequestPermissionsResult(int requestCode, @NonNull String[] permissions, @NonNull int[] grantResults) {
-    flutterView.getPluginRegistry().onRequestPermissionsResult(requestCode, permissions, grantResults);
+    flutterEngine.getPluginRegistry().onRequestPermissionsResult(requestCode, permissions, grantResults);
```

Comment:
What is the relationship between these new entities when there are multiple FlutterViews?

Previously, it was implied that each FlutterView had one FlutterNativeView (with its own registry and thus plugins) unless you explicitly told it to share a FlutterNativeView. Does that change now?

Reply:
The idea I'm working towards is that you can think of a FlutterView as being completely independent of the Flutter engine/isolate/rendering that backs it.  As such, you can attach and detach FlutterEngines to any FlutterViews at will.

The analogy I made in the PR write-up compared FlutterView to a ListView.  You can attach and detach ListAdapters whenever you want and there is no lasting relationship between a ListView and a detached ListAdapter.

Reply:
And, about plugins, I expect every FlutterEngine to have its own PluginRegistrar and plugin context.